client/manager/manager.go

```diff
@@ -0,0 +1,92 @@
+package manager
+
+import (
+	"github.com/coreos/dex/client"
+	"github.com/coreos/dex/pkg/log"
+	"github.com/coreos/dex/repo"
+	"github.com/coreos/go-oidc/oidc"
+)
+
+// Manager performs client-related "business-logic" functions on client and related objects.
+// This is in contrast to the Repos which perform little more than CRUD operations.
+type ClientManager struct {
+	clientRepo client.ClientRepo
+	begin      repo.TransactionFactory
+}
+
+type ClientManagerOptions struct {
+	// This is empty right now but will soon contain configuration information
+}
+
+func NewClientManager(clientRepo client.ClientRepo, txnFactory repo.TransactionFactory, options ClientManagerOptions) *ClientManager {
+	return &ClientManager{
+		clientRepo: clientRepo,
+		begin:      txnFactory,
+	}
+}
+
+func (m *ClientManager) New(cli client.Client) (*oidc.ClientCredentials, error) {
+	return m.clientRepo.New(nil, cli)
+}
+
+func (m *ClientManager) Get(id string) (client.Client, error) {
+	return m.clientRepo.Get(nil, id)
+}
+
+func (m *ClientManager) All() ([]client.Client, error) {
+	return m.clientRepo.All(nil)
+}
+
+func (m *ClientManager) Metadata(clientID string) (*oidc.ClientMetadata, error) {
+	c, err := m.clientRepo.Get(nil, clientID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &c.Metadata, nil
+}
+
+func (m *ClientManager) IsDexAdmin(clientID string) (bool, error) {
+	c, err := m.clientRepo.Get(nil, clientID)
+	if err != nil {
+		return false, err
+	}
+
+	return c.Admin, nil
+}
+
+func (m *ClientManager) SetDexAdmin(clientID string, isAdmin bool) error {
+	tx, err := m.begin()
+	if err != nil {
+		return err
```

Comment:
missing a bunch of rollbacks here.

A better pattern we stumbled on in more recent code was to immediately `defer rollback()`, since a rollback will have no effect (just returns ErrTxDone) for a successfully committed tx.